README.md

```diff
@@ -5,757 +5,886 @@
 }
 
-create(Person, { name: 'Stewie' })
-  .father.name.set('Peter')
-  .father.father.name.set('Mr Griffin')
-  .father.father.father.name.set('Mr Giffin Senior').state;
-// => { name: 'Stewie',
-//      father: {
-//        name: 'Peter',
-//        father: {
-//          name: 'Mr Griffin,
-//          father: {
-//            name: 'Mr Griffin Senior'
-//          }
-//        }
-//      }
-//    }
+let blog = create(Blog, {
+  posts: {
+    "1": { id: "1", title: "Hello World" },
+    "2": { id: 2, title: "Most fascinating blog in the world" }
+  }
+});
+
+blog.posts["0"].state;
+//> Post<{ id: 1, title: 'Hello World' }>
+
+blog.posts["1"].state;
+//> Post<{ id: 2, title: 'Most fascinating blog in the world' }>
+```
+
+Parameterized objects have `Object` type transitions which allow you to use `assign` and `put`.
+
+```js
+let blog2 = blog.posts.put("3", { id: 3, title: "It is only getter better" });
+
+blog2.posts["3"].state;
+//> Post<{ id: 3, title: 'It is only getter better' }>
 ```
 
-We currently do not support composition inside of arrays but we're looking into supporting it in the
-future.
+# State
+
+State is an object graph that is lazily constructred from a Microstate's type and value. It mirrors the structure of the Microstate type that was used to create it.
+
+## Building state
 
-Composed states have two default transitions `set` and `merge`.
+State is built from the root of the Microstate down, one child at a time, as your application reaches for that state. The constuction of the state is done lazily which affords the use of recursive data structures. `class` types are instantiated from their JavaScript classes.
 
-`set` will replace the state of the current microstate.
+Let's consider a fairly deep data structure:
 
 ```js
-create(Person).father.father.set({ name: 'Peter' }).state;
-// => { name: '', father: { name: 'Peter' }}
+class App {
+  shop = Shop
+}
+
+class Shop {
+  products = [Product]
+}
+
+class Product {
+  title = String;
+}
 ```
 
-`merge` will recursively merge the object into the current state.
+If we were to create this type, we'd get the following:
 
 ```js
-create(Person, { name: 'Peter' }).merge({ name: 'Taras', father: { name: 'Serge' } }).state;
-// { name: 'Taras', father: { name: 'Serge' }}
+import { create } from 'microstates';
+
+let app = create(App, {
+  shop: {
+    products: [
+      { id: 1, title: 'Sponge' }
+    ]
+  }
+});
+
+app.state instanceof App //> true
+app.state.shop instanceof Shop //> true
+app.state.shop.products instanceof Array //> true
+app.state.shop.products[0] instanceof Product //> true
+app.state.shop.products[0].title === 'Sponge' // true
+app.state.shop.products[0].id === 1 //> true
 ```
 
-## Static values
+This illustrates that the object that's created from the state property is the same structure as the Microstate itself. According to its structure, each instance is populated by it's corresponding value from the value object.
+
+## All getters are cached
```

Comment:
This is no longer true, right? Also, getters are now queries, so we may have to re-address this.